lib/spack/spack/test/cmd/env.py

```diff
@@ -833,7 +833,8 @@ def test_lockfile_spliced_specs(environment_from_manifest, install_mockery):
     # The one spec is mpileaks
     for _, spec in e2.concretized_specs():
         assert spec.spliced
-        assert spec["mpi"].satisfies(zmpi)
+        assert spec["mpi"].satisfies(f"zmpi@{zmpi.version}")
+        assert spec["mpi"].build_spec.satisfies(zmpi)
```

Comment:
I think this should also test that `spec["mpi"].build_spec.dag_hash() == zmpi.dag_hash()`.  @becker33 what do you think?

Reply:
I don't think it matters whether we test `satisfies` or dag_hash equality for concrete specs, but checking dag_hash is faster.

Reply:
somehow I missed that this IS checking the build spec. sigh.